### PR TITLE
Fix process-agent and trace-agent custom-config configuration

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Add missing `custom-config` volumeMount in `process-agent` and `trace-agent`. 
 
### Motivation

Fix `process-agent` and `trace-agent` configuration when the user provide a `customConfig`.

### Additional Notes

N/A

### Describe your test plan

Create a new DatadogAgent deployment. In the agent.spec add a `customConfig` configMap. Check
If this configMap is properly mounted in the `process-agent` and the `trace-agent`.

### CODE REVIEW

File: controllers/datadogagent/agent_test.go [hunk 1]
```diff
@@ -824,6 +872,29 @@ func defaultPodSpec() corev1.PodSpec {
...
+		Command:         []string{"trace-agent", "--config=/etc/datadog-agent/datadog.yaml"},
+		Resources:       corev1.ResourceRequirements{},
+		Ports:           []corev1.ContainerPort{{Name: "traceport", ContainerPort: 8126, Protocol: "TCP"}},
+		Env:             defaultAPMContainerEnvVars(),
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: "config", MountPath: "/etc/datadog-agent"},
+			{
+				Name:      "custom-datadog-yaml",
+				ReadOnly:  true,
+				MountPath: "/etc/datadog-agent/datadog.yaml",
```
Comment (xornivore) [inline controllers/datadogagent/agent_test.go:889]: I feel like we should have these as consts, it's repeated in another form above.

**Review by xornivore — COMMENTED** [review]
Looks good (small comment about const), thoughts about adding more tests? 😄

File: controllers/datadogagent/agent_test.go (hunk 1 above)
Comment (clamoriniere) [reply controllers/datadogagent/agent_test.go:889]: the const value exist in `api/v1alpha1`. but the idea here was to no use it in the test

Comment (clamoriniere) [reply controllers/datadogagent/agent_test.go:889]: I can add another const value in the test file.

**Review by xornivore — APPROVED** [review]

**Pull request merged**